### DISCUSSION TITLE
Valid keyboard model was not found

### DISCUSSION THREAD

**Discussion by Rai96** [4JX/L5P-Keyboard-RGB#64] [Q&A]
Hi, I have legion 5 Pro (16ACH6H), and when I run your app it says, "A valid keyboard model was not found. It may be due to a hardware error." But my RGB indeed is working. 
I just need to add, that if I am able to use your application it would be great as my RGB is stuck on the wave effect.
Thank You

**Comment by 4JX** [comment]
> Hi, I have legion 5 Pro (16ACH6H), and when I run your app it says, "A valid keyboard model was not found. It may be due to a hardware error." But my RGB indeed is working. I just need to add, that if I am able to use your application it would be great as my RGB is stuck on the wave effect. Thank You

Does vantage prompt you to install a driver when you open the rgb section?

**Reply by Rai96** [reply]
It used to before, but now the option to set different RGB modes is gone. And your software is kinda my hope :/

**Reply by 4JX** [reply]
Can you search and try to see if you have an "unknown device" (or similar) in your device manager (somewhere in the keyboard section probably).

If that's the case, try to uninstall/remove it.

**Reply by Rai96** [reply]
I checked under the keyboard, and none of them are unknown devices and then I checked all the categories too. I also reinstalled all the keyboard drivers, but after running your software, it still shows the same error.
Can you recommend me any other troubleshooting methods? If anything I want the RGB off at the moment. I'm new to coding (doing JS right now), but I don't know how can I make a script that turns off the RGB.
Any help would be thankful.

**Reply by 4JX** [reply]
Don't really have anymore help other than
- Try reinstalling vantage, the drivers or windows itself

It shouldn't be a hardware issue, but its best to rule everything else out first.

**Comment by heckert-gamedev** [comment]
The older Version (from 2021-12-11) used to work for me but with the latest system updates stopped working with a crash
```
The application panicked (crashed).
Message:  hidapi error: hid_error is not implemented yet
Location: src/main.rs:52
```

Luckily there is v 0.17 from July available now. If I run it as user I get the same message about there not being a valid keyboard.

It does work if I run the program with sudo (both versions actually).
